### ISSUE TITLE
An abstract class should not be accepted where a class type is expected

### ISSUE THREAD

**Issue by Pyprohly** [microsoft/pyright#3179]
```python
# pyright: strict

from abc import ABC, abstractmethod

class Base(ABC):
    @abstractmethod
    def get_name(self) -> str:
        raise NotImplementedError

    def print_name(self) -> None:
        print("Hello, " + self.get_name())

def greet(typ: type[Base]) -> None:
    instance = typ()
    instance.print_name()

greet(Base)  # <--
```

Mypy: `error: Only concrete class can be given where "Type[Base]" is expected`.
Pyright: No error.

I think Pyright should match Mypy’s behaviour.

<details>
<summary> Additional context </summary>
The example is from the TypeScript handbook. I was just curious what the Python type checkers had to say had to say about this.

https://www.typescriptlang.org/docs/handbook/2/classes.html#abstract-construct-signatures

```typescript
abstract class Base {
  abstract getName(): string;
 
  printName() {
    console.log("Hello, " + this.getName());
  }
}

function greet(ctor: typeof Base) {
  const instance = new ctor();
  instance.printName();
}

greet(Base)
```
</details>

**Comment by erictraut** [comment]
I disagree with mypy's behavior in this case. Pyright does emit an error if you attempt to instantiate an abstract class, but there's no reason you should not be able to pass an abstract base class as a type. It's a base class, after all, and `type[Base]` represents all subclasses of `Base`.
